### ISSUE TITLE
Big sized messages and settled_mode

### ISSUE THREAD

**Issue by ndanzi** [Azure/azure-uamqp-c#27]
Hi all,

I'm trying to send a message with a payload of 5mb to an IoTHub.
The library tells that the message has been sent successfully and no error is flagged in any callback. But the server does not receive the message because of its limitations (messages bigger then 64kb are dropped by the server). 
I set
```javascript
link_set_snd_settle_mode(linkHandle, sender_settle_mode_settled);
link_set_max_message_size(linkHandle, 5000000);
```
I supposed that the option `sender_settle_mode_settled` should force the library to tell if the message has been delivered to the endpoint but I am not able to catch any state that indicates that the message has not been delivered. in fact the send result state in the callback of the `messagesender_send` if finally set to `MESSAGE_SEND_OK`.

Is there a way to know if the message has been delivered to the endpoint?

**Comment by dcristoloveanu** [comment]
Hi @ndanzi,

Setting sender_settle_mode_settled means that the transfers are considered by the sender to be settled.
AMQP spec section  2.8.2 states:

2.8.2 Sender Settle Mode
Settlement policy for a sender.
<type name="sender-settle-mode" class="restricted" source="ubyte">
<choice name="unsettled" value="0"/>
<choice name="settled" value="1"/>
<choice name="mixed" value="2"/>
</type>
Valid Values
0 The sender will send all deliveries initially unsettled to the receiver.
1 The sender will send all deliveries settled to the receiver.
2 The sender MAY send a mixture of settled and unsettled deliveries to the receiver.

This literally means that the receiver will not report the state of the delivery as it is considered settled.

If you do want to have the delivery state be considered unsettled until the receiver confirms it as such please use the unsettled mode in the enumeration.

I would like to look a big deeper though on what happens on the server side.
Can you please provide some more details on:

The sample code you run as server.
The client code you run.
If possible can you please turn on logging and post the logs here?
Also if you could give me the commit Id you used for uamqp that will help in reproducing and making sure the issues gets fixed.

To turn on logging: 

```C
connection_set_trace(connection, true);
```

in your samples?

Thanks,
/Dan

**Comment by ndanzi** [comment]
Thank you for the reply.

I mistakenly got it backwards: with unsettled mode now actually the client fails sending the message. 
In this way I am able to catch the delivery state of the message.

For uamqp I used the Release 2016-11-17 with some paches (in order to make it work on windos ce 6.0).
I also applied the fixes of [Address Issue #24: default values for fields if they are NULL and hav…](https://github.com/Azure/azure-uamqp-c/commit/0a9588e443891a54b5094133aaa5d504c14bd235) and [Fix fields that are nmot defaulted by the generator in order to fix i…](https://github.com/Azure/azure-uamqp-c/commit/1727c0a6309710bb30921710adc6332d17ead2b3)
As a client I run the message_sender_sample provided in the samples.

The limitations on server side are due to the free account which limitates the incoming message by size

Thanks
Nicolò
